subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/rules/ComponentReplacementIntegrationTest.groovy

```diff
@@ -39,8 +39,7 @@ class ComponentReplacementIntegrationTest extends AbstractIntegrationSpec {
                 doLast {
                     println "resolved files=" + files*.name.toSorted()
                     if (!${GradleContextualExecuter.configCache}) {
-                        // Hit legacy API to trigger both result loading logic
```

Comment:
It seems like there was an explicit intent to also verify the legacy codepath in these tests. Do we not think that is necessary anymore?

Reply:
I don't think so.  That comment was added in 2021 along with the test `"handles replacement when target is a dependency of replaced"` which originally failed with CC.  

It was since patched to work with CC and every test in the class continues to pass without this entire if block present.